### ISSUE TITLE
Bug in util-linux 2.40.1 rename causes data loss

### ISSUE THREAD

**Issue by glensc** [Homebrew/homebrew-core#175757]
There's this bug:
- https://github.com/util-linux/util-linux/issues/3071

Seems to be macOS specific, but it can cause data loss because multiple files are renamed to same filename.

Will homebrew-core accept patch reverting the breaking commit?
- https://github.com/util-linux/util-linux/commit/7b67193a53209907f84a6253dcf3aceaf2eb8da1

### Verification

- [ ] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [ ] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Trying to prevent data loss for other users.

It's unclear how fast upstream will respond especially as macOS is not their intended use.

### What happened (include all command output)?

I lost some files

### What did you expect to happen?

NOT to lose files.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install util-linux
touch test_0{1,2,3,4,5}.htm
rename -v .htm .html *.htm
test -e test_01.html
test ! -e .html
```

**Comment by ryandesign** [comment]
> Will homebrew-core accept patch reverting the breaking commit?

The breaking commit fixed a different bug; reverting it would unfix it so that's probably not ideal either.

**Comment by iMichka** [comment]
From my understanding the "different bug" was a rename being ignored when there is a trailing / in the path. That's far less critical than the new bug that was introduced. I think we should revert that commit. Maybe we can wait 24h more just to see if upstream gives a more straightforward solution.

**Comment by ryandesign** [comment]
There is now an upstream commit that purports to fix the problem; I have not tested yet.

**Comment by ZhongRuoyu** [comment]
I've tested https://github.com/util-linux/util-linux/commit/ff8ee29d648111eb222612ad4251e4c3b236a389 and https://github.com/util-linux/util-linux/commit/0309a6f5ca018d83420e49e0f9d046fecdb29261 locally and verified that this fixes the issue.

**Comment by ryandesign** [comment]
Me too and I added the fix to MacPorts in macports/macports-ports@baf43ea76b241b71522cce544a8918d89cb76c17.
